File: go/config/IPv6.go
package config

import (
	"github.com/lvcdy/huawei-lte-api-go/session"
)

// IPv6 对应 config/IPv6.py。
type IPv6 struct {
	*session.ApiGroup
}

// NewIPv6 创建 IPv6 API 分组。
func NewIPv6(s *session.Session) *IPv6 {
	return &IPv6{ApiGroup: session.NewApiGroup(s)}
}

// Config 对应 ipv6/config.xml。
func (i *IPv6) Config() (map[string]interface{}, error) {
	return i.S.Get("ipv6/config.xml", nil, "config")
}

File: src/config/ipv6.rs
//! IPv6 config group (`config/IPv6.py`).

use serde_json::Value;

use crate::api::get_value;
use crate::connection::Connection;
use crate::errors::Result;

/// IPv6 config group.
pub struct Ipv6Config<'a> {
    conn: &'a Connection,
}

impl<'a> Ipv6Config<'a> {
    /// Build a group bound to `conn`.
    pub fn new(conn: &'a Connection) -> Self {
        Ipv6Config { conn }
    }

    /// `ipv6/config.xml`. IPv6 configuration.
    pub fn config(&self) -> Result<Value> {
        get_value(self.conn, "config/ipv6/config.xml")
    }
}
